### PR TITLE
NETOBSERV-635 UI Consistency: Tabs options

### PR DESCRIPTION
**Pull request by jpinsonneau**
This PR reorganize tabs options under an "advanced options" button showing an extra toolbar with:
- A display option dropdown show per view options
  - Overview
image
  - Table
image
  - Topology 
image
- Find in view for topology
image
- Manage panels / columns for `overview` & `table` views
image
- Export button for `table` / `topology` view (check [followup for overview](https://issues.redhat.com/browse/NETOBSERV-669))
image

### CODE REVIEW

**Comment by jpinsonneau** [comment]
FYI topology shortcuts will be implemented in followup https://issues.redhat.com/browse/NETOBSERV-406

**Comment by andrew-ronaldson** [comment]
> * image

This looks good, thanks @jpinsonneau 
One minor change is that the border-bottom on the tab bar ends before the "hide advanced options" text. Could it go to the end, please?

**Comment by jpinsonneau** [comment]
> This looks good, thanks @jpinsonneau One minor change is that the border-bottom on the tab bar ends before the "hide advanced options" text. Could it go to the end, please?

Sure I can find a way to fake it

**Comment by jpinsonneau** [comment]
Fixed tabs + advanced options button bottom border
image

**Comment by Amoghrd** [comment]
@jpinsonneau I see that some drop-downs most of them open in-place whereas some open to the side. Is that expected behavior? 

For example scope opens to the side
Screen Shot 2022-10-28 at 12 59 53 PM

whereas edge labels opens inplace
Screen Shot 2022-10-28 at 1 00 50 PM

**Comment by jpinsonneau** [comment]
> @jpinsonneau I see that most of the drop-downs open in-place whereas some open to the side. Is that expected behavior?

Yes @Amoghrd. When you have 2 dropdowns in the same line, I open them under the input. Else I open on the right.
This is a temporary behavior until we find something better with @andrew-ronaldson

File: web/src/components/netflow-topology/netflow-topology.tsx [hunk 1]
```diff
@@ -99,69 +99,76 @@ export const TopologyContent: React.FC<{
...
-        //go back to first match if last item is reached
-        if (lastNodeIdsFound.length === matchingNodeModels.length) {
-          lastNodeIdsFound = [];
-        }
-      } else {
-        if (lastNodeIdsFound.length === 1) {
-          //fill matching ids except last
-          lastNodeIdsFound = matchingNodeModels.map(n => n.id);
-          lastNodeIdsFound.splice(-1);
+      console.log('onSearch', searchValue, next, lastNodeIdsFound);
```
Comment (jotak) [inline web/src/components/netflow-topology/netflow-topology.tsx:118]: console.log to remove

File: web/src/components/netflow-topology/netflow-topology.tsx [hunk 2]
```diff
@@ -99,69 +99,76 @@ export const TopologyContent: React.FC<{
...
+
```
Comment (jotak) [inline web/src/components/netflow-topology/netflow-topology.tsx:119]:
```suggestion
```

File: web/src/components/netflow-topology/netflow-topology.tsx [hunk 3]
```diff
@@ -418,6 +417,27 @@ export const TopologyContent: React.FC<{
     }
   }, [selected, selectedIds]);
 
+  React.useEffect(() => {
+    console.log('search effect', searchEvent, searchHandle);
```
Comment (jotak) [inline web/src/components/netflow-topology/netflow-topology.tsx:421]:
```suggestion
```

File: web/src/components/search/search.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  updateIndicators: (count: string, indicators: ValidatedOptions) => void;
+};
+
+export const SearchComponent: React.FC<{
+  isDark?: boolean;
+  setSearchEvent: (se: SearchEvent) => void;
+  ref?: React.Ref<SearchHandle>;
+  children?: React.ReactNode;
+  // eslint-disable-next-line react/display-name
+}> = React.forwardRef((props, ref: React.Ref<SearchHandle>) => {
```
Comment (jotak) [inline web/src/components/search/search.tsx:24]: Can you elaborate what's this `forwardRef` / why it's needed?
Also, apparently there's a workaround to not have the linter warning, cf this comment: https://github.com/jsx-eslint/eslint-plugin-react/issues/2269#issuecomment-491567533

**Comment by jotak** [comment]
lgtm just maybe remove the remaining `console.log` ?

File: web/src/components/netflow-topology/netflow-topology.tsx (hunk 2 above)
Comment (jotak) [reply web/src/components/netflow-topology/netflow-topology.tsx:119]: no problem, it always happens! 😁

File: web/src/components/search/search.tsx (hunk 4 above)
Comment (jpinsonneau) [reply web/src/components/search/search.tsx:24]: Forward ref allows you to get [a ref](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-aeef245327564ecae089389ec6a41a108f749f48003cde41d9f3095da8e09464R196) from [a child component](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-aeef245327564ecae089389ec6a41a108f749f48003cde41d9f3095da8e09464R993):
https://reactjs.org/docs/forwarding-refs.html

This is usefull here to be able to propagate [search ref](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-aeef245327564ecae089389ec6a41a108f749f48003cde41d9f3095da8e09464R845-R846) and [expose custom functions](https://github.com/netobserv/network-observability-console-plugin/pull/215/files#diff-b3601c183b04e4c8caaf93d6bdf26a3dc2188ea67c8a9d36cb558bfa9dfbed65R14) to child components.
I assume we can have search features in other tabs so with that approach we will be able to reuse the same search component everywhere.

**Pull request merged**